src/MCStack.cpp

```diff
@@ -414,16 +414,23 @@ MCStack::isInDeadlock() const
     const MCTransition &nextTransitionForTid =
       this->getNextTransitionForThread(tid);
 
-    if (nextTransitionForTid.ensuresDeadlockIsImpossible())
+    if (this->getThreadDataForThread(tid).getExecutionDepth() >=
+        this->configuration.maxThreadExecutionDepth) {
+      return false;
+    }
+
+    if (nextTransitionForTid.ensuresDeadlockIsImpossible()) {
       return false;
+    }
 
     // We don't use the wrapper here (this->transitionIsEnabled)
     // because we only care about if the schedule *could* keep going:
     // we wouldn't be in deadlock if we artificially restricted the
     // threads
     if (MCTransition::transitionEnabledInState(this,
-                                               nextTransitionForTid))
+                                               nextTransitionForTid)) {
       return false;
+    }
```

Comment:
The function `MCTransition::transitionEnabledInState()` only considers the current state and not the execution depth. So we shouldn't need to add the above check. What was the problematic trace?

Reply:
Based on our discussion, the point is that sometimes at a given maxDepth, we may discover a long trace for deadlock, but if we use a larger maxDepth, then we can discover a shorter trace that uses the thread at maxDepth to a larger extent